include/sisl/cache/simple_cache.hpp

```diff
@@ -37,19 +37,34 @@ class SimpleCache {
 
 public:
     SimpleCache(const std::shared_ptr< Evictor >& evictor, uint32_t num_buckets, uint32_t per_val_size,
-                key_extractor_cb_t< K, V >&& extract_cb, Evictor::can_evict_cb_t evict_cb = nullptr) :
+                key_extractor_cb_t< K, V >&& extract_cb, Evictor::eviction_cb_t evict_cb = nullptr) :
             m_evictor{evictor},
             m_key_extract_cb{std::move(extract_cb)},
             m_map{num_buckets, m_key_extract_cb, std::bind(&SimpleCache< K, V >::on_hash_operation, this, _1, _2, _3)},
             m_per_value_size{per_val_size} {
-        m_record_family_id = m_evictor->register_record_family(std::move(evict_cb));
+        m_record_family_id = m_evictor->register_record_family(Evictor::RecordFamily{.can_evict_cb = evict_cb
+            , .post_eviction_cb = [this](const CacheRecord& record) {
+                V const value = reinterpret_cast<SingleEntryHashNode< V >*>(const_cast< CacheRecord* >(&record))->m_value;
+                K key = m_key_extract_cb(value);
+                return m_map.try_erase(key);
```

Comment:
I think it would be great if we can have a block of comment here explain why we use try_erase which calls try_lock and the purpose here is to avoid deadlock.